aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/CredentialCaches.kt

```diff
@@ -209,7 +216,14 @@ class CredentialCacheConfig(codegenContext: ClientCodegenContext) : ConfigCustom
                             #{Some}(credentials_cache),
                             #{Some}(credentials_provider),
                         ) => {
-                            resolver.config_mut().store_put(credentials_cache.create_cache(credentials_provider));
+                            let credentials_cache = credentials_cache.create_cache(credentials_provider);
+                            resolver.config_mut().store_put(credentials_cache.clone());
```

Comment:
For my understanding, why does the cache also get written back to config?

Reply:
Ah, good point. Line 220 can go away. I made it so based on my stale understanding of the cache, before the time runtime components had been introduced and the cache was picked up directly from the config bag. Now that we have runtime components, we don't need to write this back to config as long as we write back a new identity to runtime components.

Reply:
Updated in 3f09577.